common/recommender/Baseline.js

```diff
@@ -15,27 +15,134 @@ class Baseline {
   constructor(history, options = {}) {
     this.domainCounts = history.reduce(this.countDomainOccurrences, new Map());
     this.options = options;
+    // Features that are extracted from URLs and need normalization.
+    // Key 0 holds the min, key 1 holds max, using arrays for brevity.
+    this.normalizeFeatures = {
+      idf: [1, 0],
+      imageCount: [1, 0],
+      description: [1, 0],
+      pathLength: [1, 0],
+      largestImage: [1, 0],
+      queryLength: [1, 0]
+    };
   }
 
-  scoreEntry(entry) {
+  extractFeatures(entry) {
     const urlObj = URL(entry.url);
-    const host = URL(entry.url).host;
+    const host = urlObj.host;
+    const occurences = this.domainCounts.get(host) || 1;
```

Comment:
(and same typo three other places in this PR, if I ctrl+F for "occurence" on this page.)